GetStream/stream-chat-swift
# [CIS-1900] Push Notifications - multi bundle support

### 🔗 Issue Links

CIS-1900

### 🎯 Goal

Allow integrator to leverage multi-bundle feature via exposing new parameter to `addDevice` API called `provider name`.
Provider name represents the name of the push configuration created in Stream dashboard.
Push configurations are located under `Push Notifications` tab in the dashboard.
 
Multi-bundle support allows the integrator to list multiple client apps under one Stream app in the dashboard. (eg. when leveraging `Staging` vs. `Production` setup).
The default configuration will be used if no `provider name` is specified.


Available API:
APN:
```
client?.currentUserController().addDevice(.apn(token: deviceToken, providerName: "Name of the push configuration listed in Stream's Dashboard"))
```

Firebase:
```
client?.currentUserController().addDevice(.firebase(token: deviceToken, providerName: "Name of the push configuration listed in Stream's Dashboard"))
```

### 📝 Changes

* New optional `providerName` parameter can be specified for both `apn` and `firebase` integrations.
* Existing `provider` parameter in the `PushDevice` struct was renamed to `pushProvider` to gain API alignment with Android.
* In order to have a way to test the to multi-bundle support, new `DemoApp` target was added to the [integration apps repo](https://github.com/GetStream/stream-chat-swift-integration-apps/tree/multi-bundle) with bundle identifier `io.getstream.iOS.ChatDemoAppTwo`.
![appstore](https://user-images.githubusercontent.com/4924709/174339908-6e6b334c-4651-4492-8261-b181a63805ab.png)

* Additionally, a new Info section was added to our `DemoApp` to clarify which push configuration is used by our default develop app.
![Simulator](https://user-images.githubusercontent.com/4924709/174340596-f1c88dee-6641-4bbe-83aa-35c625c41277.jpg)

* `client` property inside `StreamChatWrapper` is no longer forced-cast.

# Important Note 📜
Docs will be added in separate PR, this PR contains only the  implementation part.

### 🧪 Manual Testing Notes

1. Build multi bundle app on `multi-bundle` branch in [integration apps repo](https://github.com/GetStream/stream-chat-swift-integration-apps/tree/multi-bundle)
3. Send a message to `Luke` from a different app/web
4. Verify the push notification is received.
![IMG_1087](https://user-images.githubusercontent.com/4924709/174606083-f6ace02b-0cbb-4b83-af0c-f57816611ba7.jpg)


### ☑️ Contributor Checklist

### 🎁 Meme

_Provide a funny gif or image that relates to your work on this pull request. (Optional)_
